File: vibe-coding/todo-app/frontend_web/src/modules/shared/domain/enums/ResponseKeyEnum.ts
/**
 * Claves del sobre de respuesta de la API. Gemelo de `response_key_enum.py`.
 *
 * Aqui se ve para que sirve tener el mismo enum en los dos lados: si el backend
 * cambia "data" por otra cosa, el sitio donde hay que tocar en el front es UNO.
 */
export const ResponseKeyEnum = {
    STATUS: "status",
    DATA: "data",
    ERROR: "error",
} as const;

File: vibe-coding/todo-app/backend_web/src/modules/shared/domain/enums/response_key_enum.py
from typing import final


@final
class ResponseKeyEnum:
    """Claves del sobre de respuesta que arma cada controller.

    Todas las respuestas de la API tienen la misma forma:

        exito -> {"status": 200, "data": {...}}
        error -> {"status": 404, "error": "list 7 not found"}

    El front controller lee `status` para traducirlo al codigo HTTP, asi que el
    nombre de esa clave no puede escribirse a mano en cada controller: se
    escribiria mal una vez y el fallo seria dificil de ver.

    Es una clase de constantes y no un Enum porque estas cadenas se usan como
    claves de diccionario. Con un `str, Enum` habria que escribir `.value` en cada
    uso y el codigo quedaria mas ruidoso sin ganar nada.
    """

    STATUS = "status"
    DATA = "data"
    ERROR = "error"
